jujugui/static/gui/src/app/components/budget-chart/budget-chart.js

```diff
@@ -21,32 +21,79 @@ with this program.  If not, see <http://www.gnu.org/licenses/>.
 YUI.add('budget-chart', function() {
 
   juju.components.BudgetChart = React.createClass({
+    propTypes: {
+      budgets: React.PropTypes.object
+    },
+
+    /**
+      Generate the widths for the chart bars.
+
+      @method _generateStyles
+      @param {Int} limit The allocation limit.
+      @param {Int} allocated The amount previously allocation.
+      @param {Int} newAllocations The new amount to be allocated.
+      @returns {Object} The element styles.
+    */
+    _generateStyles: function(limit, allocated, newAllocations) {
+      // Set the width of the new allocations bar to the percent of the limit.
+      var newWidth = ((newAllocations / limit * 100) || 0) + '%';
```

Comment:
Yep, if the limit is zero the percent will equal NaN, so there's a `|| 0` to catch that and display 0 instead of NaN. I've added a comment to that effect.